moment_kinetics/src/time_advance.jl

```diff
@@ -867,21 +877,16 @@ function time_advance!(pdf, scratch, t, t_input, vz, vr, vzeta, vpa, vperp, gyro
         else
             time_advance_no_splitting!(pdf, scratch, t, t_input, vz, vr, vzeta, vpa, vperp, gyrophase, z, r,
                 moments, fields, spectral_objects, advect_objects,
-                composition, collisions, geometry, boundary_distributions,
+                composition, collisions, geometry, gyroavs, boundary_distributions,
                 external_source_settings, num_diss_params, advance, fp_arrays,  scratch_dummy,
-                manufactured_source_list, i)
+                manufactured_source_list, diagnostic_checks, i)
         end
         # update the time
         t += t_input.dt
 
-        if i == t_input.nstep
-            # Ensure all output is written at the final step
-            finish_now = true
-        end
-
-        if mod(i,t_input.nwrite_moments) == 0 || mod(i,t_input.nwrite_dfns) == 0 || finish_now
+        if diagnostic_checks
             # update the diagnostic chodura condition
-            update_chodura!(moments,scratch[end].pdf,vpa,vperp,z,r,spectral_objects.r_spectral,composition,geometry,scratch_dummy,advect_objects.z_advect)
+            # update_chodura!(moments,scratch[end].pdf,vpa,vperp,z,r,spectral_objects.r_spectral,composition,geometry,scratch_dummy,advect_objects.z_advect)
 
```

Comment:
Calculation has moved to `update_derived_moments!()`, so delete commented-out version.
```suggestion
```